bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/nano/WlanThermoNanoV1CommandHandler.java

```diff
@@ -0,0 +1,266 @@
+        }
+
+        String groupId = channelUID.getGroupId();
+        List<Channel> channelList = data.getChannel();
+        System system = data.getSystem();
+        Unit<Temperature> unit = "F".equals(system.getUnit()) ? ImperialUnits.FAHRENHEIT : SIUnits.CELSIUS;
+
+        if (channelUID.getId().startsWith(CHANNEL_PREFIX)) {
+            int channelId = Integer.parseInt(groupId.substring(CHANNEL_PREFIX.length())) - 1;
+            if (channelList.size() > 0 && channelId <= channelList.size()) {
+                Channel channel = channelList.get(channelId);
+                switch (channelUID.getIdWithoutGroup()) {
+                    case CHANNEL_NAME:
+                        if (command instanceof StringType) {
+                            channel.setName(command.toFullString());
+                            return true;
+                        }
+                    case CHANNEL_MIN:
+                        if (command instanceof QuantityType) {
+                            channel.setMin(((QuantityType<?>) command).toUnit(unit).doubleValue());
+                            return true;
+                        }
+                    case CHANNEL_MAX:
+                        if (command instanceof QuantityType) {
+                            channel.setMax(((QuantityType<?>) command).toUnit(unit).doubleValue());
+                            return true;
+                        }
+                    case CHANNEL_ALARM_DEVICE:
+                        if (command instanceof OnOffType) {
+                            BigInteger value;
+                            if (command == OnOffType.ON) {
+                                value = BigInteger.valueOf(channel.getAlarm()).setBit(1);
+                            } else {
+                                value = BigInteger.valueOf(channel.getAlarm()).clearBit(1);
+                            }
+                            channel.setAlarm(value.intValue());
+                            return true;
+                        }
+                    case CHANNEL_ALARM_PUSH:
+                        if (command instanceof OnOffType) {
+                            BigInteger value;
+                            if (command == OnOffType.ON) {
+                                value = BigInteger.valueOf(channel.getAlarm()).setBit(0);
+                            } else {
+                                value = BigInteger.valueOf(channel.getAlarm()).clearBit(0);
+                            }
+                            channel.setAlarm(value.intValue());
+                            return true;
+                        }
+                    case CHANNEL_COLOR_NAME:
+                        if (command instanceof StringType) {
+                            channel.setColor(UtilNano.toHex(((StringType) command).toString()));
+                            return true;
+                        }
+                }
+            }
+        } else if (channelUID.getId().equals(CHANNEL_PITMASTER_1)) {
```

Comment:
```suggestion
        } else if (channelUID.getId().startsWith(CHANNEL_PITMASTER_1)) {
```

Reply:
That's actually a difference in the NanoV1 device, it does only provide a single Pitmaster channel.
Therefore, I can directly check for the whole id instead of checking the prefix first and then obtaining the actual channel numbers.

Reply:
But just a couple of lines down you are expecting `channelUID.getIdWithoutGroup()` to return different results? Wouldn't the `CHANNEL_PITMASTER_1` be the only result that could be returned? As the code is written now the switch statement serves no purpose.

Reply:
True, mixed up group and ID. Fixed.